### PR TITLE
Adding CCInt32 module

### PR DESCRIPTION
**Pull request by Fourchaux**
Seems to be a junior job.
Hoping not to make too many junior mistakes !

### CODE REVIEW

**Comment by c-cube** [comment]
Looks really nice, thank you. Can you add a `module Int32 = CCInt32` in the main containers.ml file?

(also, I'd like the opinion of @jpdeplaix on that!)

**Comment by hcarty** [comment]
The the infix operators be in an `Infix` module as in  http://c-cube.github.io/ocaml-containers/last/containers/CCFloat/index.html#module-Infix?  Maybe `include Infix` in the main `CCInt32` as well.

**Comment by c-cube** [comment]
@hcarty that makes sense, but it'd probably be the topic of another PR since Int64 suffers from the same flaw. I'm going to merge this as is.

File: src/core/CCInt32.ml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+let (lsl) = shift_left
+
+let (lsr) = shift_right_logical
+
+let (asr) = shift_right
+
+let equal (x:t) y = Pervasives.(=) x y
+
+let hash x = Pervasives.abs (to_int x)
```
Comment (jpdeplaix) [inline src/core/CCInt32.ml]: Same comment as https://github.com/c-cube/ocaml-containers/pull/203/files#r171272323

**Comment by jpdeplaix** [comment]
I think it should be perfect to agree to a "common" interface for those kind of module.
By common I mean:
* thinking about what we are putting into a ```Infix``` submodule
* should it be just included in the module itself ? If so, do we put everything or just a subset ?
* what sort of operation should be included in most cases ? (to_/of_ functions, that kind of thing)

I would personally be in favour of having no operators at all in the modules themselves for instance.

**Comment by c-cube** [comment]
I've been updating `CCInt` to reflect the new int32/int64 interfaces. The rationale for the current state of things (having an `Infix` module + include it) is to support the two distinct use cases:

- `open Int32.Infix` in a module that will make very heavy use of their infix operators (but still use qualified names, because an unqualified `of_string` is super evil)
- very local overloading such as `Int32.(1l + 2l * 3l <> 10l)`

**Pull request merged**